doc/glossary.rst

```diff
@@ -1247,14 +1246,6 @@ Methods
         (i.e. training and test data together) before further modelling, as
         this results in :term:`data leakage`.
 
-    ``get_feature_names``
```

Comment:
**Not actionable**: I'm surprised that Sphinx is not throwing a warning for the glossary entry not being defined. We still use it here:

https://github.com/scikit-learn/scikit-learn/blob/29f80b058fd037c8c4a22ad0638ca7d833aa264e/doc/whats_new/v1.0.rst?plain=1#L415

To be safe, should we remove the `:term:` part when referencing `get_feature_names`?

Reply:
The warnings are there:
```
2022-09-08T16:29:54.9157061Z /home/runner/work/scikit-learn/scikit-learn/doc/whats_new/v0.20.rst:827: WARNING: term not in glossary: get_feature_names
2022-09-08T16:29:54.9257794Z /home/runner/work/scikit-learn/scikit-learn/doc/whats_new/v0.20.rst:1400: WARNING: term not in glossary: get_feature_names
2022-09-08T16:29:55.3847924Z /home/runner/work/scikit-learn/scikit-learn/doc/whats_new/v0.24.rst:18: WARNING: term not in glossary: get_feature_names
2022-09-08T16:29:55.4959034Z /home/runner/work/scikit-learn/scikit-learn/doc/whats_new/v1.0.rst:414: WARNING: term not in glossary: get_feature_names
```

I checked the changelog and until now we didn't remove the markers when we removed functions or classes that where deprecated 2 releases before. When looking at the changelogs they are just not clickable links anymore. I'd say we don't bother and leave it as is, but I don't mind removing them.

Reply:
If there are warnings, I'll prefer to remove them. By removing the warnings for undefined glossary terms, it will be easier to filter for other more impactful warnings.

Reply:
I would also rather not introduce any new warning. We already have too many.

Reply:
done